modules/gapi/src/backends/oak/goakbackend.cpp

```diff
@@ -0,0 +1,691 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#include <opencv2/gapi/gkernel.hpp> // GKernelPackage
+
+#ifdef HAVE_OAK
+
+#include <cstring>
+#include <unordered_set>
+#include <algorithm> // any_of
+#include <functional> // reference_wrapper
+
+#include <api/gbackend_priv.hpp>
+#include <backends/common/gbackend.hpp>
+
+#include <opencv2/gapi/streaming/meta.hpp> // streaming::meta_tag
+
+#include "depthai/depthai.hpp"
+
+#include <opencv2/gapi/oak/oak.hpp>
+#include "oak_media_adapter.hpp"
+
+namespace cv { namespace gimpl {
+
+// Forward declaration
+class GOAKContext;
+struct OAKNodeInfo;
+
+class GOAKExecutable final: public GIslandExecutable {
+    friend class GOAKContext;
+    virtual void run(std::vector<InObj>&&,
+                     std::vector<OutObj>&&) override {
+        GAPI_Assert(false && "Not implemented");
+    }
+
+    virtual void run(GIslandExecutable::IInput &in,
+                     GIslandExecutable::IOutput &out) override;
+
+    void LinkToParentHelper(ade::NodeHandle handle,
+                            const std::vector<ade::NodeHandle>& nodes);
+
+    class ExtractTypeHelper : protected dai::Node {
+    public:
+        using Input = dai::Node::Input;
+        using Output = dai::Node::Output;
+        using InputPtr = dai::Node::Input*;
+        using OutputPtr = dai::Node::Output*;
+    };
+
+    struct OAKNodeInfo {
+        std::shared_ptr<dai::Node> node = nullptr;
+        bool processed = false;
+        std::vector<ExtractTypeHelper::InputPtr> inputs = {};
+        std::vector<ExtractTypeHelper::OutputPtr> outputs = {};
+    };
+
+    std::reference_wrapper<cv::GArg> packArg(const GArg &arg, std::vector<ExtractTypeHelper::InputPtr>& oak_ins,
+                                             std::vector<cv::GArg>& oak_wrapped_args);
+    void outArg(const RcDesc &rc, std::vector<ExtractTypeHelper::OutputPtr>& oak_outs);
+
+    const ade::Graph& m_g;
+    GModel::ConstGraph m_gm;
+    cv::GCompileArgs m_args;
+
+    std::unordered_map<ade::NodeHandle,
+                       OAKNodeInfo,
+                       ade::HandleHasher<ade::Node>> m_oak_nodes;
+
+    // Will be reworked later when XLinkIn will be introduced as input
+    std::shared_ptr<dai::node::ColorCamera> m_camera_input;
+    cv::Size m_camera_wh;
```

Comment:
Looks like this is the camera `size`

Reply:
Renamed, thanks

Reply:
There is the same one in `OAKKernelParams`